.github/workflows/docker.yml

```diff
@@ -11,20 +11,37 @@ on:
       - "verification/curator-service/**"
 
 jobs:
-  test:
+  test-dev:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Build Stack
+      - name: Build stack
         run: docker-compose -f dev/docker-compose.yml -f dev/docker-compose.ci.yml up --build -d -V
       # Containers are run detached above, which means we have to wait until they are up to produce meaningful logs.
       - name: Sleep for 30 seconds
         uses: jakejarvis/wait-action@master
         with:
           time: "30s"
-      - name: Print Logs
+      - name: Print logs
         run: docker-compose -f dev/docker-compose.yml logs
-      - name: Test Curator Service
+      - name: Fetch Curator Service sources
         run: docker-compose -f dev/docker-compose.yml exec -T curator curl --retry-connrefused --retry 20 --retry-delay 5 -f --silent --show-error http://localhost:3001/api/sources
-      - name: Test Data Service
+      - name: Test Data Service cases
         run: docker-compose -f dev/docker-compose.yml exec -T data curl --retry-connrefused --retry 20 --retry-delay 5 -f --silent --show-error http://localhost:3000/api/cases
+  test-prod:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build stack
+        run: docker-compose -f prod/docker-compose.yml -f prod/docker-compose.ci.yml up --build -d -V
+      # Containers are run detached above, which means we have to wait until they are up to produce meaningful logs.
```

Comment:
Is running detached necessary? E.g., will the action block on the process if it isn't in the background.

Sleep solution feels like it's bound to flake. But if it's the simplest solution for now, we can come back to it later.

Reply:
Yes each step must return a non 0 return code otherwise the next ones do not run this is why it is detached, I haven't experienced a flake so far so I'm hopeful, the wait is really for the logs to print something meaningful (we could also remove the logs step tbh, it helped me debug stuff when I was first setting things up but now it's not that useful)